zilliqa/tests/it/eth.rs

```diff
@@ -1031,3 +1031,29 @@ async fn get_transaction_by_index(mut network: Network) {
         .unwrap();
     assert_eq!(txn.hash, h2);
 }
+
+#[zilliqa_macros::test]
+async fn block_subscription(mut network: Network) {
+    let wallet = network.genesis_wallet().await;
+    let provider = wallet.provider();
+
+    let mut block_stream = provider.subscribe_blocks().await.unwrap();
```

Comment:
I think the `wallet` already exposes the method directly - there's no need to access the inner `provider`:
```suggestion
    let mut block_stream = wallet.subscribe_blocks().await.unwrap();
```

Reply:
Thanks!